packages/rspack/src/config/zod.ts

```diff
@@ -1176,94 +1208,98 @@ export const getRspackOptionsSchema = memoize(() => {
-		chunkIds: z
-			.enum(["natural", "named", "deterministic", "size", "total-size"])
-			.optional(),
-		minimize: z.boolean().optional(),
-		minimizer: z.literal("...").or(plugin).array().optional(),
-		mergeDuplicateChunks: z.boolean().optional(),
-		splitChunks: z.literal(false).or(optimizationSplitChunksOptions).optional(),
-		runtimeChunk: optimizationRuntimeChunk.optional(),
-		removeAvailableModules: z.boolean().optional(),
-		removeEmptyChunks: z.boolean().optional(),
-		realContentHash: z.boolean().optional(),
-		sideEffects: z.enum(["flag"]).or(z.boolean()).optional(),
-		providedExports: z.boolean().optional(),
-		concatenateModules: z.boolean().optional(),
-		innerGraph: z.boolean().optional(),
-		usedExports: z.enum(["global"]).or(z.boolean()).optional(),
-		mangleExports: z.enum(["size", "deterministic"]).or(z.boolean()).optional(),
-		nodeEnv: z.union([z.string(), z.literal(false)]).optional(),
-		emitOnErrors: z.boolean().optional(),
-		avoidEntryIife: z.boolean().optional()
-	}) satisfies z.ZodType<t.Optimization>;
+			),
+			fallbackCacheGroup: z
+				.strictObject({
+					chunks: optimizationSplitChunksChunks,
+					minSize: numberOrInfinity,
+					maxSize: numberOrInfinity,
+					maxAsyncSize: numberOrInfinity,
+					maxInitialSize: numberOrInfinity,
+					automaticNameDelimiter: z.string()
+				})
+				.partial(),
+			hidePathInfo: z.boolean(),
+			...sharedOptimizationSplitChunksCacheGroup
+		})
+		.partial() satisfies z.ZodType<t.OptimizationSplitChunksOptions>;
+
+	const optimization = z
+		.strictObject({
+			moduleIds: z.enum(["named", "natural", "deterministic"]),
+			chunkIds: z.enum([
+				"natural",
+				"named",
+				"deterministic",
+				"size",
+				"total-size"
+			]),
+			minimize: z.boolean(),
+			minimizer: z.literal("...").or(plugin).array(),
+			mergeDuplicateChunks: z.boolean(),
+			splitChunks: z.literal(false).or(optimizationSplitChunksOptions),
+			runtimeChunk: optimizationRuntimeChunk,
+			removeAvailableModules: z.boolean(),
+			removeEmptyChunks: z.boolean(),
+			realContentHash: z.boolean(),
+			sideEffects: z.enum(["flag"]).or(z.boolean()),
+			providedExports: z.boolean(),
+			concatenateModules: z.boolean(),
+			innerGraph: z.boolean(),
+			usedExports: z.enum(["global"]).or(z.boolean()),
+			mangleExports: z.enum(["size", "deterministic"]).or(z.boolean()),
+			nodeEnv: z.union([z.string(), z.literal(false)]),
+			emitOnErrors: z.boolean(),
+			avoidEntryIife: z.boolean()
+		})
+		.partial() satisfies z.ZodType<t.Optimization>;
 	//#endregion
 
 	//#region Experiments
 	const rspackFutureOptions = z.strictObject({
 		bundlerInfo: z
 			.strictObject({
-				version: z.string().optional(),
-				bundler: z.string().optional(),
-				force: z
-					.boolean()
-					.or(z.array(z.enum(["version", "uniqueId"])))
-					.optional()
+				version: z.string(),
+				bundler: z.string(),
+				force: z.boolean().or(z.array(z.enum(["version", "uniqueId"])))
 			})
+			.partial()
 			.optional()
```

Comment:
nitpick: a missing `.optional()` :)

Although changing to `.partial()` does not reduce the calls to create schema, it would be great to keep the same style and future properties adding to `rspackFuture` could avoid the `.optional()`.

Reply:
ditto for:

- `assetParserDataUrlOptions`
- `assetParserOptions`
- `cssParserOptions`
- `cssModuleParserOptions`

Reply:
Nice catch 👍 I've changed the related objects to use `partial()` (although it may only has one property currently)